src/connect.rs

```diff
@@ -397,12 +427,115 @@ impl Connect for Connector {
     }
 }
 
+//impl Connect for Connector {
+//    type Transport = Conn;
+//    type Error = BoxError;
+//    type Future = Connecting;
+//
+//    fn connect(&self, dst: Uri) -> Self::Future {
+//        let timeout = self.timeout;
+//        for prox in self.proxies.iter() {
+//            if let Some(proxy_scheme) = prox.intercept(&dst) {
+//                return with_timeout(
+//                    self.clone().connect_via_proxy(dst, proxy_scheme),
+//                    timeout,
+//                )
+//                    .boxed();
+//            }
+//        }
+//
+//        with_timeout(
+//            self.clone().connect_with_maybe_proxy(dst, false),
+//            timeout,
+//        )
+//            .boxed()
+//    }
+//}
+
+
 pub(crate) trait AsyncConn: AsyncRead + AsyncWrite {}
 impl<T: AsyncRead + AsyncWrite> AsyncConn for T {}
-pub(crate) type Conn = Box<dyn AsyncConn + Send + Sync + Unpin + 'static>;
+
+pin_project! {
+    pub(crate) struct Conn {
+        #[pin]
+        inner: Box<dyn AsyncConn + Send + Sync + Unpin + 'static>,
+        is_proxy: bool,
+    }
+}
+
+impl Connection for Conn {
+    fn connected(&self) -> Connected {
+        Connected::new().proxy(self.is_proxy)
```

Comment:
Yea to get the `HttpInfo` from the `TcpStream`, this would perhaps need to be able to call `self.inner.connected()`.

Reply:
Calling it on `inner` is impossible because it's `Box<dyn AsyncConn + Send + Sync + Unpin + 'static>`. I can access `Conneced` by calling `io.connected()` after `let io = http.call(dst).await?;`. But there is still a problem to return `Connected` from `connected` fn. There possibly two options: 

1. Save `Connected` from `io` to `Conn`, and clone it in `connected` fn, but it's not `Clone` in hyper
2. Save `HttpInfo` from `extra`to `Conn`. `HttpInfo` is `Clone` so it would be possible to construct `Connected`. But there is no accessor for `extra` field in `Connected`.

Reply:
I guess we could change the it so `trait AyncConn: AsyncRead + AsyncWrite + Connection`, and forward that way?

Reply:
I think it's boxed because it's not necessary a `TcpStream`, it can also be a `Rustls` or `NativeTls` channel. They don't implement `Connection`